fastq/compare.go

```diff
@@ -0,0 +1,135 @@
+package fastq
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+	"log"
+	"strings"
+)
+
+const debugMode = 0 //set debugMode to 1 to enable prints
+
+//AllAreEqual returns true if each Fastq entry in a slice of Fastq structs contains identical information, false otherwise.
+func AllAreEqual(alpha []Fastq, beta []Fastq) bool {
+	if len(alpha) != len(beta) {
+		if debugMode > 0 {
+			log.Printf("AllAreEqual is false. len(alpha): %d. len(beta): %d.\n", len(alpha), len(beta))
+		}
+		return false
+	}
+	for i := range alpha {
+		if !IsEqual(alpha[i], beta[i]) {
+			if debugMode > 0 {
+				log.Printf("AllAreEqual is false. Entries at index %d do not match.", i)
+			}
+			return false
+		}
+	}
+	return true
+}
+
+//IsEqual returns true if two input Fastq structs contain identical information, false otherwise.
+func IsEqual(alpha Fastq, beta Fastq) bool {
+	if CompareName(alpha, beta) != 0 {
+		if debugMode == 1 {
+			log.Printf("IsEqual: Name is not equal. Alpha: %s. Beta: %s.\n", alpha.Name, beta.Name)
+		}
+		return false
+	}
+	if CompareSeq(alpha, beta) != 0 {
+		if debugMode == 1 {
+			log.Printf("IsEqual: Seq is not equal. Alpha: %s. Beta: %s.\n", dna.BasesToString(alpha.Seq), dna.BasesToString(beta.Seq))
+		}
+		return false
+	}
+	if CompareQuals(alpha, beta) != 0 {
+		return false
+	}
+	return true
+}
+
+//CompareName compares two Fastq names for sorting or equality testing.
+func CompareName(alpha Fastq, beta Fastq) int {
+	return strings.Compare(alpha.Name, beta.Name)
```

Comment:
I think we should move away from using strings.Compare. From the godoc for strings.Compare: 

"Compare is included only for symmetry with package bytes. It is usually clearer and always faster to use the built-in string comparison operators ==, <, >, and so on."